### PR TITLE
✨Add basic playtest window

### PR DESCRIPTION
**Pull request by Speak2Erase**
- Adds a basic playtest window
  - Tries to launch `steamshim`, and then defaults on `game` 
- Includes an integrated semi-decent terminal
  - Terminal has some issues with the cursor not displaying right
  - Also has an issue with scrolling looking weird
  - The functionality is all there though 
- Also includes a terminal window based on the same terminal
  - Opens powershell on windows, or `$SHELL` on linux

Overall the integrated terminal works great on linux and windows (there's an issue with it not capturing mkxp/stock rgss's executable's output, but I'll fix that later)
It's not the best thing in the world, but it's usable, which is good enough in my eyes. 
Short of pulling in someone else's terminal (maybe i could snag wezterm-gui somehow?) and using that instead of our own it probably won't get much better without a lot of effort

### CODE REVIEW

File: src/components/top_bar.rs [hunk 1]
```diff
@@ -203,6 +203,33 @@ impl TopBar {
 
         ui.separator();
 
+        #[cfg(not(target_arch = "wasm32"))]
+        ui.add_enabled_ui(info.filesystem.project_loaded(), |ui| {
+            if ui.button("Playtest").clicked() {
+                match std::process::Command::new(
+                    info.filesystem.project_path().unwrap().join("steamshim"),
+                )
+                .stdin(std::process::Stdio::piped())
```
Comment (somedevfox) [inline src/components/top_bar.rs]: Please, avoid long paths for structures, enumerations and methods if you're using several of them from one module.
This is purely for the sake of better looking code.

```rust
use std::process;

match process::Command::new(...);
.stdin(process::Stdio::piped())
.stdout(process::Stdio::piped())
```
or
```rust
use std::process::{Stdio, Command};

Command::new(...);
Stdio::piped();
```

File: src/filesystem/filesystem_native.rs [hunk 2]
```diff
@@ -135,7 +135,7 @@ impl super::filesystem_trait::Filesystem for Filesystem {
         path: impl ToString,
     ) -> Result<(), String> {
         let mut path = PathBuf::from(path.to_string());
-        let mut original_path = path.clone().to_string_lossy().to_string();
+        let original_path = path.clone().to_string_lossy().to_string();
```
Comment (somedevfox) [inline src/filesystem/filesystem_native.rs:138]: I think it's my code, but-
We should *fail* if we can't convert any `PathBuf` to a `String`.
[`PathBuff::to_string_lossy`](https://doc.rust-lang.org/std/path/struct.Path.html#method.to_string_lossy ) replaces all invalid characters with `U+FFFD` symbol - resulting in an invalid path, resulting in *path not found error when it's used to open a file or read directory.*

**Review by somedevfox — CHANGES_REQUESTED** [review]

File: src/components/top_bar.rs (hunk 1 above)
Comment (Speak2Erase) [reply src/components/top_bar.rs]: This uhh isn't the code in the latest commit

Comment (somedevfox) [reply src/components/top_bar.rs]: Iii didn't notice, my apologies-

File: src/filesystem/filesystem_native.rs (hunk 2 above)
Comment (Speak2Erase) [reply src/filesystem/filesystem_native.rs:138]: Yeaaa I'm not sure what to do about that, but I can try and fix that later

**Pull request merged**